### PR TITLE
[mixpanel-browser] Feature/group profiles

### PR DESCRIPTION
**Pull request by dwilt**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [z] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mixpanel-browser ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mixpanel-browser@latest/))
was missing the following properties:
1. config
1. persistence
1. cookie
1. track_with_groups
1. track_pageview

<details>
<summary>as well as these 2 other properties...</summary>
<p>name_tag, toString</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5660b9297d44fd8cbc42204dfc078884de152026
</p>

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:145]: Setup methods [per docs](https://developer.mixpanel.com/docs/javascript-full-api-reference#mixpanelgroup)

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:3]: Any idea why these files weren't already prettied?

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:141]: Added group methods

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:145]: Added

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:165]: Added

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:162]: Added

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:151]: Added

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [inline types/mixpanel-browser/index.d.ts:179]: @karlos1337 why do we have individual function exports as well as [the ones defined in the `Mixpanel` interface](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46097/files#diff-55d6f62151213aea3e8e542d44ee24c3R144)?

Comment (karlos1337) [reply types/mixpanel-browser/index.d.ts:179]: Sorry @dwilt, can't say, I'm totally disconnected from this, I'd created this definitions for a project that I'm not currently maintaining and from so long 😅, maybe @RicardoRodrigues or @randak  could help.

Comment (randak) [reply types/mixpanel-browser/index.d.ts:179]: Unfortunately, I've only recently started working on the types for this library, because we use it regularly in my work. I don't have any historical context on why the definitions are structured this way. If you can test and verify whether the typing is valid without the duplication, perhaps you would be willing to create a follow up PR after this one is merged in order to make that change?

File: types/mixpanel-browser/index.d.ts
Comment (randak) [inline types/mixpanel-browser/index.d.ts:217]: This seems to be missing from the Mixpanel interface above.

File: types/mixpanel-browser/index.d.ts
Comment (randak) [reply types/mixpanel-browser/index.d.ts:179]: @karlos1337 You may want to remove yourself from the maintainers list on these definitions so you don't keep getting tagged with this 😄

File: types/mixpanel-browser/index.d.ts
Comment (randak) [reply types/mixpanel-browser/index.d.ts:3]: No idea, I joined this recently as you are doing here, and they were already like this.

**Review by randak — CHANGES_REQUESTED** [review]
Overall, this looks good, the one thing missing is the track_with_groups on the Mixpanel interface. Let me know once you've made that change and I can take another look. Regarding the duplication of types, I don't have the historical context on that but if it can be removed, I'd be happy to look at another PR for it

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [reply types/mixpanel-browser/index.d.ts:180]: I might look at firing up another PR to look at this duplicate issue.

File: types/mixpanel-browser/index.d.ts
Comment (dwilt) [reply types/mixpanel-browser/index.d.ts:218]: Fixed

**Comment by dwilt** [comment]
@randak ready for another review!

**Comment by elibarzilay** [comment]
@karlos1337 @RicardoRodrigues @randak: reping

**Review by randak — APPROVED** [review]

**Comment by orta** [comment]
Alright, I'm going to merge this as it's been reviewed already 👍🏻

**Pull request merged**
